enterprisemediawiki/meza
# Enable HTTPS by default, provide SAML setup script

Sets up HTTPS and, optionally, SAML authentication.

## Testing

* `curl -LO https://raw.githubusercontent.com/enterprisemediawiki/meza/saml/scripts/install.sh`
* `sudo bash install.sh`
* Use `saml` branch
* Perform normal tests. Make sure you use https to access via browser.
* Run `sudo bash saml.sh` and follow the prompts. There are many prompts which I'm not going to explain here. SAML will not function on a development VM anyway (though perhaps we should create a testing identity provider (our own Launchpad) for future testing). Any prompts you enter will be used in setting up files in `/opt/meza/simplesamlphp`: `./config/config.php`, `./config/authsources.php`, and `./metadata/saml20-idp-remote.php`. Additionally, `httpd.conf` and `LocalSettings.php` are modified.

## Changes

* Parsoid accesses MediaWiki's PHP API via port 8142 now. Special handling in httpd.conf.
* Firewall opens both http and https ports 80 and 443 (for future forwarding of http to https)
* Attempted to make http forward to https, which worked for that purpose but broke Elasticsearch. This has been disabled for now and will be solved later.
* httpd load mod_ssl
* http vs https option removed from initial prompts in install.sh
* Prompts added to install.sh for setting up self-signed certificate. Though initial certificate is self-signed, should be easy to drop in proper certificates later. 
* Script `saml.sh` installs SimpleSamlPhp and MW extension SimpleSamlAuth. Also modified httpd.conf to include alias for simplesamlphp.

## Issues

* Closes #111 
* Closes #285
